components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java

```diff
@@ -76,6 +78,19 @@ public void startRoute(String id) throws Exception {
         camelContext.getRouteController().startRoute(id);
     }
 
+    public String resetRoute(String id) throws Exception {
+        ManagedRouteMBean resetRoute = camelContext.getManagedRoute(id, ManagedRouteMBean.class);
+        if (resetRoute != null) {
+            resetRoute.reset(true);
+        } else {
+            String description = "The provided route id does not exists";
+            ResponseEntity<?> objectResponseEntity = ResponseEntity.status(HttpStatus.NO_CONTENT).body(description);
+            ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
@vrlgohel ah okay, yeah we can return a json text message response too. I guess that is how others my do it since the endpoint indicate json content-type (I guess having to use separate for error codes may just be annoying).